File: view/lib/src/model/process.py


class Process():
    id = -1
    deadline = -1
    time_init = -1
    ttf = -1
    finish = False

    def __init__(self, id, time, ttf, deadline):
        self.id = id
        self.time_init = time
        self.deadline = deadline
        self.ttf = ttf

    def getId(self):
        return self.id
    
    def getDeadline(self):
        return self.deadline
    
    def getTimeInit(self):
        return self.time_init

    def getTimeToFinish(self):
        return self.ttf

    def updateTimeToFinish(self, quantum):
        self.ttf -= quantum

    def isFinish(self):
        return self.finish

    def setFinishedTrue(self):
        self.finish = True

File: view/lib/src/model/process.dart
class Process {
  var id = -1;
  var deadline = -1;
  var timeInit = -1;
  var ttf = -1;

  Process(id, time, ttf, deadline) {
    this.id = id;
    this.deadline = deadline;
    this.timeInit = time;
    this.ttf = ttf;
  }

  int getId() {
    return this.id;
  }

  int getDeadline() {
    return this.deadline;
  }

  int getTimeInit() {
    return this.timeInit;
  }

  int getTtf() {
    return this.ttf;
  }

  void setId(int newId) {
    this.id = newId;
  }

  void setDeadline(int newDeadline) {
    this.deadline = newDeadline;
  }

  void setTimeInit(int newTimeInit) {
    this.timeInit = newTimeInit;
  }

  void setTtf(int newTtf) {
    this.ttf = newTtf;
  }

  void updateTtf(int quantum) {
    this.ttf -= quantum;
  }
}
